src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -248,6 +225,49 @@ void ExecuteWithAbi (string [] supportedAbis, string apkInputPath, string apkOut
 						count = 0;
 					}
 				}
+
+				HashSet<ulong> deletedEntries = new HashSet<ulong> ();
```

Comment:
Why is this a `ulong` instead of a `long`, when `.EntryCount` is `long`?

Reply:
@grendello - do you know if there's a rationale for that? maybe EntryCount can be -1 in some case?

Reply:
As far as I remember, the rationale for `EntryCount` being `long` was some sort of compatibility with BCL's `ZipArchive` (which doesn't have `EntryCount` but has a collection which has a signed integer `Count` property). At the same time, I always avoid using signed integers whenever a size or an index is concerned - on the notion that we have no negative indices and we have no negative sizes (or at least - they make no sense). Thus using an unsigned integer limits the size of the error domain (index can be too big, but can't be too small - one less thing that can go wrong)